pkg/common/resources/resources.go

```diff
@@ -395,12 +395,26 @@ func subNonNegative(left, right *Resource) (*Resource, string) {
 	}
 	return out, message
 }
-
-// Check if smaller fitin larger, negative values will be treated as 0
-// A nil resource is treated as an empty resource (zero)
+// Check if smaller fits in larger, negative values will be treated as 0
+// Types not defined in the larger resource are considered 0 values for Quantity
+// A nil resource is treated as an empty resource (all types are 0)
 func FitIn(larger, smaller *Resource) bool {
```

Comment:
I think we should rename the two FitIn functions: `func FitIn(larger, smaller *Resource) bool` and` func (r *Resource) FitIn(smaller *Resource) bool` to a name what reflects the difference between them. Having the same name is easy to mix them, also while reading the code is not obvious what will happen with the undefined resources.

Reply:
I thought about that but could not really think of a good name, that is why I left it.

The difference is in the description of the function: type is not defined becomes 0, type is not defined becomes max Quantity (i.e. max int64) 
Since Quantity can never be larger and thus will always fit, the value check is skipped:
```
 if skipUndef && !ok {
  continue
}
```
implements the difference between the two.

Reply:
What about `FitInSkipUndef` for the one where we handle it as max, and left the other one as it is

Reply:
The skip undefined points to how it was implemented. It is faster than setting the value to max int64 and doing the comparison. I was trying to capture the max in the name and had `FitInMaxUndef` as my early choice.

Reply:
I am fine with `FitInMaxUndef` as well, just we need to make the difference obvious between the two methods.

Reply:
updated the name and fixed some typos in the comments